### ISSUE TITLE
what's the purpose of the function "createTree"?

### ISSUE THREAD

**Issue by yuegui2004** [Leoid/MySQL-Injection-Exfiltration-Optimization#2]
I found that it seems the function "createTree" does not work, in the main function you create three arrays with the ascii of 0-9,a-z,A-Z;  I put the ascii into three different arrays directly and I don't need the function createTree, and it still works. So I want to know what the purpose of the function "createTree" is ?

**Comment by m4p1e** [comment]
lol ,  same issue as me

**Comment by Leoid** [comment]
Hey guys, 
Try to execute only the CreateTree method aside on an array and check the result.

**Comment by m4p1e** [comment]
i mean our CreateTree didnt do its work，the good thing u do is that before guessing each character, you first judge their scope.

**Comment by Leoid** [comment]
That's right, the best case is O(Log(n)) and if the character lets say is 9 so it's the worst case then it's gonna be O(n).
Best Regards,

**Comment by m4p1e** [comment]
hmmmm,   for example the [0-9] , the probability of each character appearing is the same  , so it not important about its 0  or  9. right?

**Comment by m4p1e** [comment]
> That's right, the best case is O(Log(n)) and if the character lets say is 9 so it's the worst case then it's gonna be O(n).
> Best Regards,

i was wrong:( u are right !
